src/GafferScene/CollectPrimitiveVariables.cpp

```diff
@@ -0,0 +1,259 @@
+}
+
+const Gaffer::StringVectorDataPlug *CollectPrimitiveVariables::suffixesPlug() const
+{
+	return getChild<StringVectorDataPlug>( g_firstPlugIndex + 0 );
+}
+
+Gaffer::StringPlug *CollectPrimitiveVariables::targetVariablesPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 1 );
+}
+
+const Gaffer::StringPlug *CollectPrimitiveVariables::targetVariablesPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 1 );
+}
+
+Gaffer::BoolPlug *CollectPrimitiveVariables::isVariationPlug()
+{
+	return getChild<BoolPlug>( g_firstPlugIndex + 2 );
+}
+
+const Gaffer::BoolPlug *CollectPrimitiveVariables::isVariationPlug() const
+{
+	return getChild<BoolPlug>( g_firstPlugIndex + 2 );
+}
+
+void CollectPrimitiveVariables::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	SceneElementProcessor::affects( input, outputs );
+
+	if(
+		input == inPlug()->objectPlug() ||
+		input == suffixesPlug() ||
+		input == targetVariablesPlug()
+	)
+	{
+		outputs.push_back( isVariationPlug() );
+		outputs.push_back( outPlug()->objectPlug() );
+	}
+}
+
+void CollectPrimitiveVariables::hash( const Gaffer::ValuePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	SceneElementProcessor::hash( output, context, h );
+	if( output == isVariationPlug() )
+	{
+		if( !( filterValue( context ) & IECore::PathMatcher::ExactMatch ) )
+		{
+			h = IECore::MurmurHash();
+			h.append( false );
+			return;
+		}
+
+		h.append( inPlug()->objectPlug()->hash() );
+
+		Context::EditableScope scope( context );
+		ConstStringVectorDataPtr suffixesData = suffixesPlug()->getValue();
+		const vector<string> &suffixes = suffixesData->readable();
+
+		for( vector<string>::const_iterator it = suffixes.begin(), eIt = suffixes.end(); it != eIt; ++it )
+		{
+			scope.set( g_collectPrimitiveVariablesSuffix, *it );
+			h.append( inPlug()->objectPlug()->hash() );
+		}
+	}
+}
+
+void CollectPrimitiveVariables::compute( Gaffer::ValuePlug *output, const Gaffer::Context *context ) const
+{
+	if( output == isVariationPlug() )
+	{
+		bool isVariation = false;
+
+		if( filterValue( context ) & IECore::PathMatcher::ExactMatch )
+		{
+			IECore::MurmurHash inObjectHash = inPlug()->objectPlug()->hash();
+			IECore::ConstObjectPtr inObject = inPlug()->objectPlug()->getValue( &inObjectHash );
+
+			Context::EditableScope scope( context );
+			ConstStringVectorDataPtr suffixesData = suffixesPlug()->getValue();
+			const vector<string> &suffixes = suffixesData->readable();
+
+			for( vector<string>::const_iterator it = suffixes.begin(), eIt = suffixes.end(); it != eIt; ++it )
+			{
+				scope.set( g_collectPrimitiveVariablesSuffix, *it );
+				IECore::MurmurHash collectObjectHash = inPlug()->objectPlug()->hash();
+
+				if( collectObjectHash != inObjectHash )
+				{
+					IECore::ConstObjectPtr collectObject = inPlug()->objectPlug()->getValue( &collectObjectHash );
+					if( collectObject != inObject )
```

Comment:
Did you mean to compare pointers here, rather than pointees?

Reply:
Yep, oops.